source/collections/VectorList.ooc

```diff
@@ -123,11 +123,10 @@ VectorList: class <T> {
 		for (i in 0 .. this count)
 			this[i] = function(this[i])
 	}
-	map: func <T, S> (function: Func(T) -> S) -> This<S> {
+	map: func <S> (function: Func(T) -> S) -> This<S> {
 		result := This<S> new(this count)
-		for (i in 0 .. this count) {
+		for (i in 0 .. this count)
 			result add(function(this[i]))
-		}
 		result
 	}
 	fold: func <T, S> (function: Func(T, S) -> S, initial: S) -> S {
```

Comment:
It's still the same problem: ```error Not enough info to resolve return type S of function call```. Which could be solved by qualifying ```S: Class```, but that is not acceptable in my opinion.